server/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -211,9 +211,9 @@ public final int hashCode() {
      */
     static Object maybeConvertToBytesRef(Object obj) {
         if (obj instanceof String) {
-            return BytesRefs.toBytesRef(obj);
+            return BytesRefs.checkIndexableLength(BytesRefs.toBytesRef(obj));
```

Comment:
Do you think it is worth checking the length even before converting String to BytesRef? Or it makes the code too complex?

Reply:
There are complexities in converting between java String lengths and UTF8 encoded bytes which makes me think this is more failsafe.